XenCert/StorageHandler.py

```diff
@@ -2981,6 +2981,327 @@ def MPConfigVerificationTests(self):
+            for host in host_disable_MP_list:
+                StorageHandlerUtil.disable_multipathing(self.session, host)
+
+        except Exception, e:
+            Print("      There was an exception while performing pool consistency tests. Exception: %s. Please check the log for details." % str(e))
+            displayOperationStatus(False)
+            retVal = False
+
+            try:
+                # Try cleaning up here
+                if sr_ref != None:
+                    Print("      Destroy the SR.")
+                    StorageHandlerUtil.DestroySR(self.session, sr_ref)
+                checkPoint += 1
+
+            except Exception, e:
+                Print("Could not cleanup the SR created during testing, SR: %s. Exception: %s. Please destroy the SR manually." % (sr_ref, str(e)))
+                displayOperationStatus(False)
+
+            Print("   -> Disable multipathing on hosts. ")
+            for host in host_disable_MP_list:
+                StorageHandlerUtil.disable_multipathing(self.session, host)
+
+        XenCertPrint("Checkpoints: %d, totalCheckPoints: %s" % (checkPoint, totalCheckPoints))
+        return (retVal, checkPoint, totalCheckPoints)
+
+
+    def FunctionalTests(self):
+        retVal = True
+        sr_ref = None
+        _trash = None
+        checkPoints = 0
+        totalCheckPoints = 0
+        testFileCreated = False
+        testDirCreated = False
+        mountCreated = False
+        testSRCreated = False
+
+        mountpoint = '/mnt/XenCertTest-' + commands.getoutput('uuidgen')
+
+        totalCheckPoints += 5
+        try:
+            # 1. Display various exports from the server for verification by the user.
+            # XXX For the time being, this test will remain inactive. It will
+            # probably be implemented with CIFSSR.probe()
+            #Print("DISCOVERING SHARES FROM THE SPECIFIED TARGET")
+            #Print(">> This test probes the specified CIFS server and displays the ")
+            #Print(">> various paths exported for verification by the user. ")
+            try:
+                #cmd = ["smbclient", "-L", self.storage_conf['server'], "-U", "%s%%%s" % (self.storage_conf['username'], self.storage_conf['password'])]
+                #list =  util.pread2(cmd).split('\n')
+                #Print("%s%%%s" % (self.storage_conf['username'], self.storage_conf['password']))
+                #if len(list) > 0:
+                #    Print("   %-50s" % 'Share')
+                #for val in list:
+                #    if len(val.split()) > 0:
+                #        Print("   %-50s" % val.split()[0])
+                #displayOperationStatus(True)
+                checkPoints += 1
+            except Exception, e:
+                Print("   - Failed to display shares on server: %s. Exception: %s" % (self.storage_conf['server'], str(e)))
+                raise e
+
+            # 2. Verify CIFS target by mounting as local directory
```

Comment:
Is this required as a separate test if we are creating an SR in the next test anyway?